sympy/plotting/plot.py

```diff
@@ -2135,7 +2138,36 @@ def check_arguments(args, expr_len, nb_of_free_symbols):
             i = len(args) + 1
 
         exprs = Tuple(*args[:i])
-        free_symbols = list(set().union(*[e.free_symbols for e in exprs]))
+        # get all variables
+        variables = set().union(*[e.free_symbols for e in exprs])
+        free_symbols = list(variables)
+        # check arguments
+        assert all([isinstance(rang, Tuple) for rang in args[expr_len:]]), \
+            "Ranges should be in the form (symbol, a, b)"
+        # get symbols from all ranges
+        specify_symbols = set([s[0] for s in args[expr_len:]])
+
+        style_ranges = ",  {} = range(*)"*len(specify_symbols)
+        expr_ranges = style_ranges.format(*specify_symbols)[3:]
+
+        all_symbols = variables | specify_symbols
+        if len(all_symbols) > len(variables):
+            msg = "There be a variable ranged that isn´t part of any ecuation!"
+            raise NameError(msg)
+
+        if len(all_symbols) > nb_of_free_symbols:
+            if specify_symbols == set():
+                msg =  "There must be  ONLY %s (%s) parameters from %s"%(
```

Comment:
Is there really a need for  get_text(nb_of_free_symbols), nb_of_free_symbols ?
I don't think there is a need to write `one(1)` or `Two(2)`. and what happens when it's > 2 ? It will show something like `there must be only  (3) parameters ...`.

```python
msg = "At most %s symbols should be used out of the given: %s"(nb_of_free_symbols, all_symbols) 
```
Something like this might be better for the message.